README.md
# Factorio Mod: SimpleQualityOfLifeResearch
Simple quality of life researches without being bloaty or cheaty by providing additional inventory and steel axe researches, and player reach researches. Updated to 1.1.x by Walkman100

This is [SimpleQualityOfLifeResearch](https://mods.factorio.com/mod/SimpleQualityofLifeResearch) updated to 1.1.x, with reach researches added.


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.0.9
  Features:
    - Added player reach distance research & setting
---------------------------------------------------------------------------------------------------
Version: 1.0.8
  Changes:
    - Added science pack researches as prerequisites to QOL researches
---------------------------------------------------------------------------------------------------
Version: 1.0.7
  Features:
    - Updated to 1.1.x
    - Fixed Research icon sizes for 1.1.x
---------------------------------------------------------------------------------------------------
Version: 1.0.6
  Balance:
    - Updated to 0.18
    - Removed Logistic Slot Research as it's no longer needed
    - Now uses default pickaxe artwork
---------------------------------------------------------------------------------------------------
Version: 1.0.5
  Balance:
    - Adjusted science costs to be similar to vanilla researches
---------------------------------------------------------------------------------------------------
Version: 1.0.4
  Features:
    - Added settings to disable researches
---------------------------------------------------------------------------------------------------
Version: 1.0.3
  Features:
    - Added proper changelog
---------------------------------------------------------------------------------------------------
Version: 1.0.2
  Features:
    - Cleaned up some researches
---------------------------------------------------------------------------------------------------
Version: 1.0.1
  Balance:
    - Adjusted logistic slot costs
---------------------------------------------------------------------------------------------------
Version: 1.0.0
  Features:
    - Initial Release


prototypes/technology/inventory.lua
if settings.startup["inventory-upgrades"].value then
    data:extend({
        {
            type = "technology",
            name = "toolbelt",
            icon_size = 256,
            icon = "__base__/graphics/technology/toolbelt.png",
            prerequisites = {
                "logistic-science-pack"
            },
            effects = {
                {
                    type = "character-inventory-slots-bonus",
                    modifier = 10
                }
            },
            unit = {
                count = 100,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-k-m"
        },
        {
            type = "technology",
            name = "toolbelt-2",
            icon_size = 256,
            icon = "__base__/graphics/technology/toolbelt.png",
            prerequisites = {
                "chemical-science-pack",
                "toolbelt"
            },
            effects = {
                {
                    type = "character-inventory-slots-bonus",
                    modifier = 10
                }
            },
            unit = {
                count = 200,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-k-m"
        },
        {
            type = "technology",
            name = "toolbelt-3",
            icon_size = 256,
            icon = "__base__/graphics/technology/toolbelt.png",
            prerequisites = {
                "production-science-pack",
                "toolbelt-2"
            },
            effects = {
                {
                    type = "character-inventory-slots-bonus",
                    modifier = 10
                }
            },
            unit = {
                count = 300,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-k-m"
        },
        {
            type = "technology",
            name = "toolbelt-4",
            icon_size = 256,
            icon = "__base__/graphics/technology/toolbelt.png",
            prerequisites = {
                "utility-science-pack",
                "toolbelt-3"
            },
            effects = {
                {
                    type = "character-inventory-slots-bonus",
                    modifier = 10
                }
            },
            unit = {
                count = 500,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1},
                    {"utility-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-k-m"
        }
    })
end


prototypes/technology/mining-speed.lua
if settings.startup["mining-speed-upgrade"].value then
    data:extend({
        {
            type = "technology",
            name = "steel-axe",
            icon_size = 256,
            icon = "__base__/graphics/technology/steel-axe.png",
            prerequisites = {
                "steel-processing"
            },
            effects = {
                {
                    type = "character-mining-speed",
                    modifier = 1
                }
            },
            unit = {
                count = 50,
                ingredients = {
                    {"automation-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "steel-axe-2",
            icon_size = 256,
            icon = "__base__/graphics/technology/steel-axe.png",
            prerequisites = {
                "chemical-science-pack",
                "steel-axe"
            },
            effects = {
                {
                    type = "character-mining-speed",
                    modifier = .5
                }
            },
            unit = {
                count = 200,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "steel-axe-3",
            icon_size = 256,
            icon = "__base__/graphics/technology/steel-axe.png",
            prerequisites = {
                "production-science-pack",
                "steel-axe-2"
            },
            effects = {
                {
                    type = "character-mining-speed",
                    modifier = .5
                }
            },
            unit = {
                count = 300,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "steel-axe-4",
            icon_size = 256,
            icon = "__base__/graphics/technology/steel-axe.png",
            prerequisites = {
                "utility-science-pack",
                "steel-axe-3"
            },
            effects = {
                {
                    type = "character-mining-speed",
                    modifier = .5
                }
            },
            unit = {
                count = 500,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1},
                    {"utility-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        }
    })
end


prototypes/technology/reach.lua
if settings.startup["player-reach-upgrade"].value then
    local function get_effects(modifier)
        return {
            {
                type = "character-build-distance",
                modifier = modifier
            },
            {
                type = "character-reach-distance",
                modifier = modifier
            },
            {
                type = "character-resource-reach-distance",
                modifier = modifier
            },
            {
                type = "character-item-drop-distance",
                modifier = modifier
            }
        }
    end

    data:extend({
        {
            type = "technology",
            name = "player-reach",
            icons = {
                {
                    icon = "__core__/graphics/icons/entity/character.png",
                    icon_size = 64
                },
                {
                    icon = "__core__/graphics/icons/technology/effect-constant/effect-constant-range.png",
                    icon_size = 64
                }
            },
            prerequisites = {
                "logistic-science-pack"
            },
            effects = get_effects(2),
            unit = {
                count = 100,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "player-reach-2",
            icons = {
                {
                    icon = "__core__/graphics/icons/entity/character.png",
                    icon_size = 64
                },
                {
                    icon = "__core__/graphics/icons/technology/effect-constant/effect-constant-range.png",
                    icon_size = 64
                }
            },
            prerequisites = {
                "chemical-science-pack",
                "player-reach"
            },
            effects = get_effects(2),
            unit = {
                count = 200,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "player-reach-3",
            icons = {
                {
                    icon = "__core__/graphics/icons/entity/character.png",
                    icon_size = 64
                },
                {
                    icon = "__core__/graphics/icons/technology/effect-constant/effect-constant-range.png",
                    icon_size = 64
                }
            },
            prerequisites = {
                "production-science-pack",
                "player-reach-2"
            },
            effects = get_effects(2),
            unit = {
                count = 300,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        },
        {
            type = "technology",
            name = "player-reach-4",
            icons = {
                {
                    icon = "__core__/graphics/icons/entity/character.png",
                    icon_size = 64
                },
                {
                    icon = "__core__/graphics/icons/technology/effect-constant/effect-constant-range.png",
                    icon_size = 64
                }
            },
            prerequisites = {
                "utility-science-pack",
                "player-reach-3"
            },
            effects = get_effects(2),
            unit = {
                count = 500,
                ingredients = {
                    {"automation-science-pack", 1},
                    {"logistic-science-pack", 1},
                    {"chemical-science-pack", 1},
                    {"production-science-pack", 1},
                    {"utility-science-pack", 1}
                },
                time = 30
            },
            upgrade = true,
            order = "c-c-a"
        }
    })
end
